### DISCUSSION TITLE
"You should enable it in browser flags too" what is it ?

### DISCUSSION THREAD

**Discussion by haoadoreorange** [darkreader/darkreader#8569] [Q&A] [answered]
Hiii,

I tried to enable it on restricted sites in the new UI, and there's a small msg saying "You should enable it in browser flag too". But I'm not sure what is it and how to do it, can't seem to find anything about it either. 

Do you have any information on this ?

**Comment by Gusted** [comment, marked as answer]
Chromium: Enable chrome://flags/#extensions-on-chrome-urls 
Firefox: Emtpy `extensions.webextensions.restrictedDomains` in `about:config`

**Reply by haoadoreorange** [reply]
Thanks ! Maybe add the info somewhere in the FAQ ?

**Reply by rwp0** [reply]
I emptied from `extensions.webextensions.restrictedDomains` and even added to addon.mozilla.org to Site List, but still it doesn't work with Mozilla websites on Firefox.

**Reply by ghost** [reply]
The site list is a list of sites on which it is disabled, so the site you want to make dark should _not_ be in there.
- Enable the addon on restricted pages (as described in https://github.com/darkreader/darkreader/discussions/8569#discussioncomment-4231931)
- Set `extensions.webextensions.restrictedDomains` to an empty string in about:config (according to this comments parent comment)
- Set `extensions.quarantinedDomains.enabled` to false in about:config (according to https://support.mozilla.org/en-US/kb/quarantined-domains)
For me, FF 117.0.1-1, it works. FF Developer is currently at 118.0b7-1, so not far away. As Mozilla themselves say, it only has some features of the beta. So breaking a feature that is controlled by about:config should be unlikely.

**Reply by rwp0** [reply]
Did all of this now, and removed from Site List. No results.
I think Firefox Developer Edition is different.

**Reply by ghost** [reply]
One of the most important steps I forgot to mention is, of course, restarting the browser but that's kinda obvious. You could try switching browser (to FF stable, not Chrome ofc) 117.

**Reply by rwp0** [reply]
I did restart with:

```
about:restartrequired
```

Don't want to swtich 🙂

**Comment by galantra** [comment]
Dark Reader still doesn't do anything to privileged pages. I guess it is impeded by an internal setting of the extension – because _Dark Background and Light Text_ is able to apply its dark mode.

**Reply by Gusted** [reply]
What's your question?

**Reply by galantra** [reply]
Sorry, let me clarify.

If I open Dark Reader on a privileged page like https://addons.mozilla.org/, it doesn't make any changes to the page. The add-on still says "This page is protected by browser", even though `extensions.webextensions.restrictedDomains` is empty.

So, my comment is just a notice that the suggested solution doesn't work. It seems to be a problem with the add-on, because there are other add-ons that work on privileged pages.

**Reply by Gusted** [reply]
Yes that's correct, this solution was given in the context that the user already enabled this setting within Dark Reader. 

- Click Dark Reader icon.
- Click Dev tools (in the bottom-right corner).
- Click Preview new design.
- Click Settings -> Site list -> Enable restricted pages.

**Reply by galantra** [reply]
Thank you very much! I got it to work. I wasn't aware of those settings. :)

**Reply by rwp0** [reply]
Did all these too and restarted, doesn't work with the latest Firefox Developer edition that I'm using (at least).

**Comment by Tiagoquix** [comment]
**Edit: see https://github.com/darkreader/darkreader?tab=readme-ov-file#enable-dark-reader-on-restricted-pages-on-mozilla-firefox for the updated instructions.**

**Reply by rwp0** [reply]
Will try, thanks 🙏🏻

**Reply by divStar** [reply]
Tried it with Firefox 119 (**not** "Developer Edition") on M1 (Apple Silicon) and it worked for `https://addons.mozilla.org/`, which previously would not allow Dark Reader to work properly.
It's only after all the mentioned steps, that it started working and even then the Dark Reader somehow "detected a dark theme" on that page - with the main view being white...

It's awesome, that there is a workaround. I'd suggest linking your post @Tiagoquix around the same place you can enable Dark Reader for restricted websites.

Thank you for the instructions! 😄

**Reply by e-t-l** [reply]
These are the steps that worked.

**Reply by CajonCalhoun** [reply]
Thank you kindly. Your solution works =]

**Reply by Tiagoquix** [reply]
**Edit: see https://github.com/darkreader/darkreader?tab=readme-ov-file#enable-dark-reader-on-restricted-pages-on-mozilla-firefox for the updated instructions.**

**Comment by Tiagoquix** [comment]
**Edit: see https://github.com/darkreader/darkreader?tab=readme-ov-file#enable-dark-reader-on-restricted-pages-on-mozilla-firefox for the updated instructions.**

**Reply by divStar** [reply]
I am sorry, but I am deciding for myself what I want or don't want to revert. Usually those of us, who willingly modify our configurations, know of the risks or inform ourselves about them at the least.
Why hasn't Mozilla provided a more secure way of doing this?

I do agree, that the average person shouldn't disable this setting. But since many people on here aren't your average Joe, we prefer for the darkreader to work properly.

**Reply by Tiagoquix** [reply]
**Edit: see https://github.com/darkreader/darkreader?tab=readme-ov-file#enable-dark-reader-on-restricted-pages-on-mozilla-firefox for the updated instructions.**

**Reply by divStar** [reply]
Ah, thank you for clarification! Much appreciated!

**Comment by student13371** [comment]
Tried the following changes:

    Enable the option to activate Dark Reader on restricted pages via the extension's menu.
    Go to firefox about:config.
    Set extensions.webextensions.restrictedDomains to an empty value.
    Create extensions.webextensions.addons-restricted-domains@mozilla.com.disabled with boolean as type and set its value to true.
    Create privacy.resistFingerprinting.block_mozAddonManager with boolean as type and set its value to true.
    Restart Firefox.

As of 2024-12-26, https://addons.mozilla.org/en-GB/firefox/extensions remains white, but not dark.

**Reply by Tiagoquix** [reply]
**See https://github.com/darkreader/darkreader?tab=readme-ov-file#enable-dark-reader-on-restricted-pages-on-mozilla-firefox for the updated instructions.**

**Comment by student13371** [comment]
I followed step by step as per your link above:

...................................................................................

Step 1: change Dark Reader's settings.

    Click on the Dark Reader extension icon.
    Click on the Dev tools button (in the bottom-right corner).
    Go to Advanced and click on the Preview new design button.
    Close the developer tools window and click on the Dark Reader extension icon again.
    Go to Settings -> Advanced and enable the Enable on restricted pages setting.

Step 2: change Firefox's settings.

    Type about:config in the address bar and press Enter.
        A warning page may appear. Click Accept the Risk and Continue to proceed.
    Search for and set extensions.webextensions.restrictedDomains to an empty value (if the preference does not exist, create it with String as the type).
    Set privacy.resistFingerprinting.block_mozAddonManager to true (if the preference does not exist, create it with Boolean as the type).

After changing the necessary settings for both Dark Reader and Firefox, reload the desired page(s).
...................................................................................

After doing the above, the last step which was needed to make https://addons.mozilla.org and other pages in this domain black was to:
- Click on the Dark Reader extension icon.
- Click on addons.mozilla.org inside Dark Reader, because Dark Reader was disabled for this domain with red font still claiming that **This page is protected by browser**. After clicking on addons.mozilla.org inside Dark Reader, black theme was successfully activated for https://addons.mozilla.org 


Thank you for your help, [Tiagoquix](https://github.com/Tiagoquix)!
